client/rpc/debug/src/lib.rs

```
@@ -349,27 +351,26 @@ where
 		};
 
 		// Known ethereum transaction hashes.
-		let eth_tx_hashes: Vec<_> = statuses
-			.unwrap()
-			.iter()
-			.map(|t| t.transaction_hash)
-			.collect();
+		let eth_tx_hashes: Vec<_> = statuses.iter().map(|t| t.transaction_hash).collect();
 
 		// If there are no ethereum transactions in the block return empty trace right away.
 		if eth_tx_hashes.is_empty() {
 			return Ok(Response::Block(vec![]));
 		}
 
-		// Get the extrinsics.
-		let ext = blockchain.body(reference_id).unwrap().unwrap();
+		// Get block extrinsics.
+		let exts = blockchain
+			.body(reference_id)
+			.map_err(|e| internal_err(format!("Fail to read blockchain db: {:?}", e)))?
+			.unwrap_or_default();
```

Hmm, why not also return an error for this second layer Result/Option?